src/oic/oic/__init__.py

```diff
@@ -1034,11 +1034,16 @@ def provider_config(self, issuer, keys=True, endpoints=True,
         if r.status_code == 200:
             try:
                 pcr = response_cls().from_json(r.text)
+            except ValueError as e:
+                _err_txt = "Faulty provider config response: {}".format(e)
```

Comment:
Cannot we do this for every exception and not just `ValueError`?

Reply:
Sure, as long as it is changed to a per Exception clause if r.text becomes interesting for any other exception.